### PR TITLE
Add minipccontroller example

### PR DESCRIPTION
**Pull request by ywh114**
Add minipccontroller example
Add selfcheck packet type

### CODE REVIEW

File: examples/minipc/LatencyTest.cc [hunk 1]
```diff
@@ -61,14 +61,13 @@ void RM_RTOS_Default_Task(const void* argument) {
 
   auto minipc_session = communication::MinipcPort();
 
-  communication::chassis_data_t chassis_data; // this has to be the data type that has the maximum size
+  communication::selfcheck_data_t selfcheck_data; // this has to be the data type that has the maximum size
```
Comment (ywh114) [inline examples/minipc/LatencyTest.cc:64]: Fix comment

File: examples/minipc/MiniPCController.cc [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+    uint32_t flags = osThreadFlagsWait(RX_SIGNAL, osFlagsWaitAll, osWaitForever);
+    if (flags & RX_SIGNAL) {
+      length = uart->Read(&data);
+      minipc_session.ParseUartBuffer(data, length);
+
+      recv_cmd_id = minipc_session.GetCmdId();
+      status_data = minipc_session.GetStatus();
+
+      switch (recv_cmd_id) {
+        // TODO: Move everything into header
```
Comment (ywh114) [inline examples/minipc/MiniPCController.cc:93]: Remove comment

File: examples/minipc/MiniPCController.cc [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+            if (status_data->mode == 1){
+                selfcheck_data.mode = 1;
+                selfcheck_data.debug_int = status_data->debug_int;
+
+                minipc_session.Pack(packet_to_send, (void*)&selfcheck_data, communication::SELFCHECK_CMD_ID);
+                uart->Write(packet_to_send, minipc_session.GetPacketLen(communication::SELFCHECK_CMD_ID));
+            } else if (status_data->mode == 2){
+                // Respond with ID:
+                //  BRD: 129 (129 - 127 = 2)
+                //  See minipc.py for details
```
Comment (ywh114) [inline examples/minipc/MiniPCController.cc:113]: Use config.py instead

**Review by ywh114 — COMMENTED** [review]
fix

File: examples/minipc/LatencyTest.cc (hunk 1 above)
Comment (ywh114) [reply examples/minipc/LatencyTest.cc:64]: done

File: examples/minipc/MiniPCController.cc (hunk 3 above)
Comment (ywh114) [reply examples/minipc/MiniPCController.cc:113]: done

**Review by neilghose1 — APPROVED** [review]
Builds, looks good, ready for merge

**Pull request merged**